tests/components/broadlink/test_device.py

```diff
@@ -20,16 +20,13 @@
 async def test_device_setup(hass):
     """Test a successful setup."""
     device = get_device("Office")
-    mock_api = device.get_mock_api()
-    mock_entry = device.get_mock_entry()
-    mock_entry.add_to_hass(hass)
 
-    with patch("broadlink.gendevice", return_value=mock_api), patch.object(
+    with patch.object(
         hass.config_entries, "async_forward_entry_setup"
     ) as mock_forward, patch.object(
         hass.config_entries.flow, "async_init"
     ) as mock_init:
-        await hass.config_entries.async_setup(mock_entry.entry_id)
+        mock_api, mock_entry = await device.setup_entry(hass)
```

Comment:
We don't want to call integration internal code directly. Just call core interfaces and possibly library callbacks.

Reply:
this is from a test mock iirc

Reply:
Right, thanks!